3-networks/modules/restricted_shared_vpc/main.tf

```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Shared VPC configuration
+ *****************************************/
+
+locals {
+  vpc_type                   = "shared"
+  vpc_label                  = (var.vpc_label != "" ? var.vpc_label : "restricted")
+  vpc_name                   = "${var.environment_code}-${local.vpc_type}-${local.vpc_label}"
+  network_name               = "vpc-${local.vpc_name}"
+  restricted_googleapis_cidr = "199.36.153.4/30"
+  subnets = [
+    for subnet in var.subnets : {
+      subnet_name           = "sb-${local.vpc_name}-${subnet.subnet_region}"
+      subnet_ip             = subnet.subnet_ip
+      subnet_region         = subnet.subnet_region
+      subnet_private_access = subnet.subnet_private_access
+      subnet_flow_logs      = subnet.subnet_flow_logs
+      description           = subnet.description
+    }
+  ]
+
+  temp_secondary_ranges = [
```

Comment:
Is there a reason why we need to do this manipulation as opposed to following the same interface as networks module?